src/plugins/discover/public/application/angular/doc_table/components/table_row.ts

```diff
@@ -94,6 +95,10 @@ export function createTableRowDirective($compile: ng.ICompileService) {
         createSummaryRow($scope.row);
       });
 
+      $scope.$watch('shouldRerender', () => {
```

Comment:
there's a `$scope.$watchMulti` before, you could add `shouldRerender` since it's executing the same code